src/main/java/org/spongepowered/api/service/event/EventManager.java

```diff
@@ -48,10 +59,17 @@
     void unregister(Object obj);
 
     /**
+     * Un-registers all event handlers of a plugin.
+     *
+     * @param plugin The plugin instance
+     */
+    void unregisterPlugin(Object plugin);
```

Comment:
Conflicts with the other `unregister` method, and if given a plugin instance we wouldn't be able to tell if it's supposed to unregister the plugin class only or all plugin listeners.